micrometer-core/src/test/java/io/micrometer/core/instrument/binder/kafka/KafkaMetricsTest.java

```diff
@@ -0,0 +1,155 @@
+            metrics.put(metricName, metric);
+            return metrics;
+        };
+        KafkaMetrics kafkaMetrics = new KafkaMetrics(supplier);
+        MeterRegistry registry = new SimpleMeterRegistry();
+        //When
+        kafkaMetrics.checkAndBindMetrics(registry);
+        //Then
+        assertThat(registry.getMeters()).hasSize(1);
+        //When
+        kafkaMetrics.checkAndBindMetrics(registry);
+        //Then
+        assertThat(registry.getMeters()).hasSize(1);
+    }
+
+    @Test void shouldAddNewMetersWhenMetricsChange() {
+        //Given
+        Map<MetricName, KafkaMetric> metrics = new LinkedHashMap<>();
+        Supplier<Map<MetricName, ? extends Metric>> supplier = () -> {
+            MetricName metricName = new MetricName("a0", "b0", "c0", new LinkedHashMap<>());
+            KafkaMetric metric = new KafkaMetric(this, metricName, new Value(), new MetricConfig(), Time.SYSTEM);
+            metrics.put(metricName, metric);
+            return metrics;
+        };
+        KafkaMetrics kafkaMetrics = new KafkaMetrics(supplier);
+        MeterRegistry registry = new SimpleMeterRegistry();
+        //When
+        kafkaMetrics.checkAndBindMetrics(registry);
+        //Then
+        assertThat(registry.getMeters()).hasSize(1);
+        //Given
+        MetricName metricName = new MetricName("a1", "b1", "c1", new LinkedHashMap<>());
+        KafkaMetric metric = new KafkaMetric(this, metricName, new Value(), new MetricConfig(), Time.SYSTEM);
+        metrics.put(metricName, metric);
+        //When
+        kafkaMetrics.checkAndBindMetrics(registry);
+        //Then
+        assertThat(registry.getMeters()).hasSize(2);
+    }
+
+    @Test void shouldNotAddAppInfoMetrics() {
+        //Given
+        Map<MetricName, KafkaMetric> metrics = new LinkedHashMap<>();
+        Supplier<Map<MetricName, ? extends Metric>> supplier = () -> {
+            MetricName metricName = new MetricName("a0", "b0", "c0", new LinkedHashMap<>());
+            KafkaMetric metric = new KafkaMetric(this, metricName, new Value(), new MetricConfig(), Time.SYSTEM);
+            metrics.put(metricName, metric);
+            MetricName appInfoMetricName =
+                    new MetricName("a1", KafkaMetrics.METRIC_GROUP_APP_INFO, "c0",
+                            new LinkedHashMap<>());
+            KafkaMetric appInfoMetric =
+                    new KafkaMetric(this, appInfoMetricName, new Value(), new MetricConfig(), Time.SYSTEM);
+            metrics.put(appInfoMetricName, appInfoMetric);
+            return metrics;
+        };
+        KafkaMetrics kafkaMetrics = new KafkaMetrics(supplier);
+        MeterRegistry registry = new SimpleMeterRegistry();
+        //When
+        kafkaMetrics.checkAndBindMetrics(registry);
+        //Then
+        assertThat(registry.getMeters()).hasSize(1);
+        //When
+        kafkaMetrics.checkAndBindMetrics(registry);
```

Comment:
True. I've fixed this but using a set instead of map and reference result from supplier directly instead of adding names at the end to compare complete results.